mem3_helper.py

```diff
@@ -13,22 +13,45 @@
 import backoff
 import os
 
+class PeerDiscoveryException(Exception):
+    pass
+
 def construct_service_record():
     # Drop our Pod's unique identity and replace with '_couchdb._tcp'
     return os.getenv('SRV_RECORD') or '.'.join(['_couchdb', '_tcp'] + socket.getfqdn().split('.')[1:])
 
 @backoff.on_exception(
     backoff.expo,
     dns.resolver.NXDOMAIN,
-    max_tries=10
+    max_tries=15
+)
+@backoff.on_exception(
+    backoff.expo,
+    PeerDiscoveryException,
+    max_tries=15
```

Comment:
I've noticed you bumped up the retries number, did you observe that `10` is not sufficient?

Reply:
Yes. When backoff limit exceeded, stateful set assembler fails and K8s restarts its container, which can unnecessarily delay Helm release creation.